app/models/api_action.rb

```diff
@@ -73,7 +74,7 @@ def redirect;end
   def serve_file;end
 
   def request_headers
-    headers = custom_headers.to_json.gsub('SECRET_BEARER_TOKEN', bearer_token)
+    headers = custom_headers_evaluated.gsub('SECRET_BEARER_TOKEN', bearer_token)
```

Comment:
@alis-khadka does this allow dynamic headers here? (see below) -- if so can you add in-line documentation
![Screenshot from 2022-07-09 16-21-59](https://user-images.githubusercontent.com/35935196/178121403-993a48dd-a0e9-4eea-9c1c-93f73e72ee76.png)